### PR TITLE
HLA-1248: Bug fixes and enhancements for the segmentation catalog wrt the RickerWavelet kernel

### PR DESCRIPTION
**Pull request by mdlpstsci**
Resolves [HLA-1248](https://jira.stsci.edu/browse/HLA-1248)

Closes #

This first update to the PR is to address outright bugs when generating the RickerWavelet kernel.   In essence, a proper sigma (vs the FWHM) is now provided to the RickerWavelet2DKernel constructor, and the normalization is handled in a new method, _ricker_matched_kernel_, written by R.L. White where the normalization causes the RickerWavelet core to match the Gaussian core.

**Checklist for maintainers**
- [ ] added entry in `CHANGELOG.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

File: drizzlepac/haputils/catalog_utils.py [hunk 1]
```diff
@@ -1645,15 +1646,16 @@ def identify_sources(self, **pars):
                 log.info("")
                 log.info("The segmentation map contains big sources/islands or a large source fraction of segments.")
                 log.info("Using RickerWavelet2DKernel to generate an alternate segmentation map.")
-                rw2dk = RickerWavelet2DKernel(self.image.kernel_fwhm,
-                                                    x_size=self._rw2d_size,
-                                                    y_size=self._rw2d_size)
-                rw2dk.normalize()
+
+                # Convert the FWHM into a sigma value
+                rw_sigma = self.image.kernel_fwhm * gaussian_fwhm_to_sigma
```
Comment (rlwastro) [inline drizzlepac/haputils/catalog_utils.py:1651]:
```suggestion
                rw_sigma = self.image.kernel_fwhm / gaussian_fwhm_to_sigma
```

File: drizzlepac/haputils/catalog_utils.py [hunk 2]
```diff
@@ -1708,17 +1710,18 @@ def identify_sources(self, **pars):
                         # kernel when the background type changes
                         g2d_kernel = self.image.kernel
 
-                        rw2dk = RickerWavelet2DKernel(self.image.kernel_fwhm,
-                                                            x_size=self._rw2d_size,
-                                                            y_size=self._rw2d_size)
-                        rw2dk.normalize()
+                        rw_sigma = self.image.kernel_fwhm * gaussian_fwhm_to_sigma
```
Comment (rlwastro) [inline drizzlepac/haputils/catalog_utils.py:1713]:
```suggestion
                        rw_sigma = self.image.kernel_fwhm / gaussian_fwhm_to_sigma
```

**Review by rlwastro — CHANGES_REQUESTED** [review]
The sigma value for the kernel should be smaller than the fwhm.  The `astropy.stats.gaussian_sigma_to_fwhm` constant = ~2.35, and so the fwhm should be divided by that value to get sigma.

File: drizzlepac/haputils/catalog_utils.py (hunk 1 above)
Comment (mdlpstsci) [reply drizzlepac/haputils/catalog_utils.py:1651]: I believe this is coded correctly "as is".  The **gaussian_fwhm_to_sigma**=0.424660.  In contrast, the **gaussian_sigma_to_fwhm**=2.354820.

Comment (rlwastro) [reply drizzlepac/haputils/catalog_utils.py:1651]: > I believe this is coded correctly "as is". The **gaussian_fwhm_to_sigma**=0.424660. In contrast, the **gaussian_sigma_to_fwhm**=2.354820.

You are correct!  I was looking at the wrong value -- did not realize there were two constants there.  I will delete both of those change suggestions.

**Review by rlwastro — APPROVED** [review]

**Comment by mdlpstsci** [comment]
@rlwastro 
For some reason, your approval did not "stick".  Can you please approve again?

**Review by rlwastro — APPROVED** [review]

**Comment by zacharyburnett** [comment]
it looks like the macos runner doesn’t have HDF5 in the environment:
https://github.com/spacetelescope/drizzlepac/actions/runs/8837772821/job/24380878354?pr=1791#step:10:37
```
      .. ERROR:: Could not find a local HDF5 installation.
```
I assume that installing pytables via Conda might solve that, I’ll make up a PR

**Comment by zacharyburnett** [comment]
In the meantime, I'll force-merge this

**Pull request merged**